src/ledger/mod.rs

```diff
@@ -105,6 +105,20 @@ impl Ledger {
         Ok(())
     }
 
+    #[instrument(name = "ledger.utxo_dropped", skip(self, tx))]
+    pub async fn utxo_dropped(
+        &self,
+        tx: Transaction<'_, Postgres>,
+        tx_id: LedgerTransactionId,
+        params: UtxoDroppedParams,
```

Comment:
PR looks good!
Now try to accept the `detected_tx_id: LedgerTransactionId` here (instead of params) and reconstruct the params from the detected transaction.
Also the detected_tx_id should become the `correlation_id` of this transaction.

There are examples of how to do this in other places in this file. Not exactly that - but similar. Take a look around.